workspaces/ui-v2/src/optic-components/diffs/render/AskForCommitMessage.tsx

```diff
@@ -29,14 +37,28 @@ export default function AskForCommitMessage(props: { hasChanges: boolean }) {
 
   const pendingEndpointsCount = pendingEndpoints.filter((i) => i.staged).length;
 
-  const handleSave = () => {
-    //@todo decide if every endpoint (changes/pending) gets a batch commit id.
+  const handleSave = async () => {
+    // TODO figure out how to attach a commit message to match id
```

Comment:
So there's a commit message in the UI, but I'm not sure where this should be used? I don't see anywhere in mutations / projections where we can attach this to?

Reply:
looks like we still need to expand `applyCommands(commands: [JSON]): AppliedCommandsResult` to handle the commitMessage

it was hardcoded in spectacle mutation. 
```
      applyCommands: async (parent: any, args: any, context: any) => {
        const batchCommitId = uuidv4();
        const events = await opticContext.specRepository.listEvents();
        try {
          const newEventsString = opticContext.opticEngine.try_apply_commands(
            JSON.stringify(args.commands),
            JSON.stringify(events),
            batchCommitId,
+            'proposed changes'
          );
```

Reply:
Ahh i see - I'll also need to add a commit message into the documentation contribution flow. I'll also address this in a follow up PR